math/mathcore/inc/Fit/FitUtil.h

```diff
@@ -449,29 +449,16 @@ namespace FitUtil {
             return std::accumulate(objs.begin(), objs.end(), T{});
          };
 
-#ifndef R__USE_IMT
-         (void)nChunks;
-
-         // If IMT is disabled, force the execution policy to the serial case
-         if (executionPolicy == ::ROOT::EExecutionPolicy::kMultiThread) {
-            Warning("FitUtil::EvaluateChi2", "Multithread execution policy requires IMT, which is disabled. Changing "
-                                             "to ::ROOT::EExecutionPolicy::kSequential.");
-            executionPolicy = ::ROOT::EExecutionPolicy::kSequential;
-         }
-#endif
-
          T res{};
-         if (executionPolicy == ::ROOT::EExecutionPolicy::kSequential) {
-            ROOT::TSequentialExecutor pool;
-            res = pool.MapReduce(mapFunction, ROOT::TSeq<unsigned>(0, data.Size()/vecSize), redFunction);
-#ifdef R__USE_IMT
-         } else if (executionPolicy == ::ROOT::EExecutionPolicy::kMultiThread) {
-            ROOT::TThreadExecutor pool;
+         if (executionPolicy == ::ROOT::EExecutionPolicy::kMultiThread ||
+             executionPolicy == ::ROOT::EExecutionPolicy::kSequential) {
+            ROOT::TExecutor executor(executionPolicy);
             auto chunks = nChunks != 0 ? nChunks : setAutomaticChunking(data.Size() / vecSize);
-            res = pool.MapReduce(mapFunction, ROOT::TSeq<unsigned>(0, data.Size() / vecSize), redFunction, chunks);
-#endif
+            res = executor.MapReduce(mapFunction, ROOT::TSeq<unsigned>(0, data.Size() / vecSize), redFunction, chunks);
          } else {
-            Error("FitUtil::EvaluateChi2", "Execution policy unknown. Avalaible choices:\n ::ROOT::EExecutionPolicy::kSequential (default)\n ::ROOT::EExecutionPolicy::kMultiThread (requires IMT)\n");
+            Error("FitUtil::EvaluateChi2", "Execution policy not supported for the Fit. Avalaible choices:\n "
```

Comment:
What exactly is happening here to the code ? This is an internal function, is it possible to get an invalid policy ? 
Maybe instead of error message is better doing a fatal or an assert

Reply:
I will need to doublecheck this. I am not allowing multiprocessing, which could be the other policy to make it through.

Reply:
It is possible. In this test we are passing the execution policy as a parameter, for example:  https://github.com/root-project/root/blob/064f7f7bbaab06052c204507f3ac4bd6fbe0034b/math/mathcore/test/fit/testLogLExecPolicy.cxx#L194

And an execution policy that should not be supported and is not implemented, on top of that :thinking:

Reply:
Just checked and the Muptiprocess part of the test is never called

Reply:
So, to summarize: it is possible to get an invalid policy (ROOT::EExecutionPolicy::kMultiProcess), which is not supported at this point, and the LoglExecPolicy test needs its MP functions removed